src/features/content-explorer/content-explorer/ContentExplorer.js

```diff
@@ -492,9 +498,11 @@ class ContentExplorer extends Component {
             listWidth,
             listHeight,
             searchInputProps,
+            autoSelectedItems,
             ...rest
         } = this.props;
         const { isInSearchMode, foldersPath, selectedItems, isSelectAllChecked } = this.state;
+        const allSelectedItems = { ...selectedItems, ...autoSelectedItems };
```

Comment:
is the internal state for the initial render out of sync with the UI? is that okay? is this prop intended to be used in combination with `initialSelectedItems` for those cases?

https://github.com/box/box-ui-elements/blob/529f7c54def5dd885842ab7e33f2200b4f1e5fff/src/features/content-explorer/content-explorer/ContentExplorer.js#L176

Reply:
I feel like there are other cases where it's possible for the internal state to be out of sync with the UI if the user tries to deselect an item from `autoSelectedItems`

Reply:
The prop autoSelectedItems is meant for allowing component to have certain items selected without user having to click on those after initial render. So ideal usecase would be if we want to trigger selecting certain items as part of an action/event. That said, if autoSelectedItems has items along with autoSelectedItems then both of them would appear selected. Perhaps we want to make it clear in description that these two should be use exclusively? 
If user tries to deselect an item from autoSelectedItems then we should expect parent component to handle updating value of prop autoSelectedItems based on their requirement.